javascript-algorithms-and-data-structures/learn-intermediate-fetch-promises-and-async-await-by-building-an-fcc-forum-leaderboard/index.js

```diff
@@ -24,108 +18,114 @@ const categories = {
   560: { category: "Backend Development", className: "backend" },
 };
 
-const forumCategories = (id) => {
-  return categories.hasOwnProperty(id)
-    ? `<a class="category ${categories[id]["className"]}" target="_blank" href="${FORUM_CATEGORY_URL}${categories[id]["className"]}/${id}">${categories[id]["category"]}</a>`
-    : `<a class="category general" target="_blank" href="${FORUM_CATEGORY_URL}general/1">General</a>`;
-};
+async function fetchData() {
+  const res = await fetch(FORUM_LATEST);
+  const data = await res.json();
+  showLatestPosts(data);
+}
+fetchData();
 
-const showTime = (time) => {
-  const lastPost = new Date(time);
-  const currentTime = new Date();
+const forumCategories = (id) => {
+  let selectedCategory = {};
 
-  if (currentTime.getUTCHours() === lastPost.getUTCHours()) {
-    return `${currentTime.getMinutes() - lastPost.getUTCMinutes()}m`;
+  if (categorieList.hasOwnProperty(id)) {
+    const { category, className } = categorieList[id];
+    selectedCategory.className = className;
+    selectedCategory.category = category;
   } else {
-    return `${currentTime.getHours() - lastPost.getHours()}h`;
+    selectedCategory.className = "General";
+    selectedCategory.category = "General";
+    id = 1;
   }
+
+  const url = `${FORUM_CATEGORY_URL}${selectedCategory.className}/${id}`;
+  const linkText = selectedCategory.category;
+  const linkClass = `category ${selectedCategory.className}`;
+  const linkTarget = "_blank";
+
+  return `<a href="${url}" class="${linkClass}" target="${linkTarget}">${linkText}</a>`;
 };
 
-const countingViews = (userViews) => {
-  let count = 0;
-  let currViews = userViews;
-  while (currViews !== 0) {
-    currViews = Math.floor(currViews / 10);
-    count++;
+function showTime(time) {
+  const currentTime = new Date();
+  const lastPost = new Date(time);
+
+  const timeDifference = currentTime - lastPost;
+  const MS_PER_MINUTE = 1000 * 60;
+
+  const minutesAgo = Math.floor(timeDifference / MS_PER_MINUTE);
+  const hoursAgo = Math.floor(minutesAgo / 60);
+  const daysAgo = Math.floor(hoursAgo / 24);
+
+  if (minutesAgo < 60) {
+    return `${minutesAgo}m ago`;
+  }
+  if (hoursAgo < 24) {
+    return `${hoursAgo}h ago`;
   }
+  return `${daysAgo}d ago`;
+}
 
-  if (count >= 4) {
-    return `${userViews.toString().slice(0, -3)}k`;
-  } else {
-    return userViews;
+function countingViews(views) {
+  const thousands = Math.floor(views / 1000);
+
+  if (views >= 1000) {
+    return `${thousands}k`;
   }
+  return views;
+}
+
+const displayUsers = (posters, users) => {
```

Comment:
This function returns user avatar images, so `avatars` might be a better name:

```suggestion
const avatars = (posters, users) => {
```

`users` or `userImages` might also work.

Reply:
used avatars